tsconfig.json is not loaded when `root` is specified while `tsconfig.json` is not

Imagine I'm using the following configuration with `vite-plugin-checker`:

```ts
import { defineConfig } from 'vite';
import { checker } from 'vite-plugin-checker';

export default defineConfig(env => {
    return {
        plugins: [
            react(),
            // ...
            checker({
                typescript: {
                    root: 'src',
                    // tsconfigPath: 'tsconfig.json',
                },
            }),
        ],
    };
});
```

If I specify `root` without `tsconfigPath`, when executing `vite build`, I'll see errors like this

```
src/App.tsx(133,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/App.tsx(134,13): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/App.tsx(135,17): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
...
```

tsc acts as if there is no `tsconfig.json` specified at all. However, if I specify `tsconfigPath` explicitly (by uncommenting the code above), everything works fine.

Suspectedly, could it be something wrong with the following code snippet?

https://github.com/fi3ework/vite-plugin-checker/blob/ec4366de72365ed9dd773aa754a800291e403590/packages/vite-plugin-checker/src/checkers/typescript/main.ts#L152-L164

--- a/.changeset/good-cherries-jog.md
+++ b/.changeset/good-cherries-jog.md
@@ -1,0 +1,5 @@
+---
+"vite-plugin-checker": patch
+---
+
+Able to resolve tsconfig when only root specified in build mode, as well as vue-tsc.

--- a/packages/vite-plugin-checker/src/checkers/typescript/main.ts
+++ b/packages/vite-plugin-checker/src/checkers/typescript/main.ts
@@ -144,20 +144,23 @@ export class TscChecker extends Checker<'typescript'> {
       build: {
         buildBin: (config) => {
           if (typeof config.typescript === 'object') {
-            const { root, tsconfigPath, buildMode } = config.typescript
+            const { root = '', tsconfigPath = '', buildMode } = config.typescript
 
             // Compiler option '--noEmit' may not be used with '--build'
-            let args = [buildMode ? '-b' : '--noEmit']
+            const args = [buildMode ? '-b' : '--noEmit']
 
             // Custom config path
-            if (tsconfigPath) {
-              const fullConfigPath = root ? path.join(root, tsconfigPath) : tsconfigPath
+            let projectPath = ''
+            if (root || tsconfigPath) {
+              projectPath = root ? path.join(root, tsconfigPath) : tsconfigPath
+            }
 
+            if (projectPath) {
               // In build mode, the tsconfig path is an argument to -b, e.g. "tsc -b [path]"
               if (buildMode) {
-                args = args.concat([fullConfigPath])
+                args.push(projectPath)
               } else {
-                args = args.concat(['-p', fullConfigPath])
+                args.push('-p', projectPath)
               }
             }
 

--- a/packages/vite-plugin-checker/src/checkers/vueTsc/main.ts
+++ b/packages/vite-plugin-checker/src/checkers/vueTsc/main.ts
@@ -148,13 +148,17 @@ export class VueTscChecker extends Checker<'vueTsc'> {
       build: {
         buildBin: (config) => {
           if (typeof config.vueTsc === 'object') {
-            const { root, tsconfigPath } = config.vueTsc
+            const { root = '', tsconfigPath = '' } = config.vueTsc
 
             let args = ['--noEmit']
             // Custom config path
-            if (tsconfigPath) {
-              const fullConfigPath = root ? path.join(root, tsconfigPath) : tsconfigPath
-              args = args.concat(['-p', fullConfigPath])
+            let projectPath = ''
+            if (root || tsconfigPath) {
+              projectPath = root ? path.join(root, tsconfigPath) : tsconfigPath
+            }
+
+            if (projectPath) {
+              args.push('-p', projectPath)
             }
 
             return ['vue-tsc', args]

